p2p/providers/peerstore_provider/rpc/provider.go

```diff
@@ -37,6 +38,8 @@ type rpcPeerstoreProvider struct {
 	rpcClient *rpc.ClientWithResponses
 }
 
+// TECHDEBT(#810): refactor to be consistent with `persistencePeerstoreProvider`
+// (i.e. `NewRPCPeerstoreProvider` calls `rpcPeerstoreProvider#Create()`.
```

Comment:
Given that we have `func Create` in our factory functions, I think all `func New` should be deprecated & removed.